bundles/org.openhab.binding.modbus.studer/src/main/java/org/openhab/binding/modbus/studer/internal/StuderHandler.java

```diff
@@ -0,0 +1,432 @@
+        if (comms != null) {
+            return;
+        }
+
+        ModbusEndpointThingHandler slaveEndpointThingHandler = getEndpointThingHandler();
+        if (slaveEndpointThingHandler == null) {
+            @SuppressWarnings("null")
+            String label = Optional.ofNullable(getBridge()).map(b -> b.getLabel()).orElse("<null>");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    String.format("Bridge '%s' is offline", label));
+            logger.debug("No bridge handler available -- aborting init for {}", label);
+            return;
+        }
+        comms = slaveEndpointThingHandler.getCommunicationInterface();
+        if (comms == null) {
+            @SuppressWarnings("null")
+            String label = Optional.ofNullable(getBridge()).map(b -> b.getLabel()).orElse("<null>");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    String.format("Bridge '%s' not completely initialized", label));
+            logger.debug("Bridge not initialized fully (no endpoint) -- aborting init for {}", this);
+            return;
+        }
+    }
+
+    /**
+     * Remove the endpoint if exists
+     */
+    private void unregisterEndpoint() {
+        // Comms will be close()'d by endpoint thing handler
+        comms = null;
+    }
+
+    private synchronized void unregisterPollTasks() {
+        if (pollTasks.isEmpty()) {
+            return;
+        }
+        logger.debug("Unregistering polling from ModbusManager");
+        @Nullable
+        ModbusCommunicationInterface mycomms = comms;
+        if (mycomms != null) {
+            for (PollTask t : pollTasks) {
+                mycomms.unregisterRegularPoll(t);
+            }
+            pollTasks.clear();
+        }
+    }
+
+    /**
+     * Register poll task
+     * This is where we set up our regular poller
+     */
+    @SuppressWarnings("null")
+    private synchronized void registerPollTask(int registerNumber) {
+        if (pollTasks.size() >= registers.length) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            throw new IllegalStateException("New pollTask invalid");
+        }
+        @Nullable
+        ModbusCommunicationInterface mycomms = comms;
+        @Nullable
+        StuderConfiguration studerConfig = config;
+        if (studerConfig == null || mycomms == null) {
+            throw new IllegalStateException("registerPollTask called without proper configuration");
+        }
+
+        logger.debug("Setting up regular polling");
+
+        ModbusReadRequestBlueprint request = new ModbusReadRequestBlueprint(studerConfig.slaveAddress,
+                ModbusReadFunctionCode.READ_INPUT_REGISTERS, registerNumber, 2, studerConfig.maxTries);
+        long refreshMillis = studerConfig.refreshSec * 1000;
+        PollTask pollTask = mycomms.registerRegularPoll(request, refreshMillis, 1000, result -> {
```

Comment:
This will throw an IllegalArgumentException, if the user doesn't set `studerConfig.refreshSec` or set it to 0. You could add a sanity check in `initialize()`.

Reply:
I hadn't noticed, now I change it into required on config description XML files. Refresh setted to 0 isn't handled by `registerRegularPoll` method but I based the code on the SunSpec binding. I tested it and now I add a sanity check > 0

Reply:
I hadn't noticed, now I change it into required on config description XML files with a minimum value of 1. Refresh setted to 0 isn't handled by `registerRegularPoll` method but I based the code on the SunSpec binding. I tested it and I think a check is not needed